tsl/src/compression/compression.c

```diff
@@ -729,6 +730,8 @@ row_compressor_update_group(RowCompressor *row_compressor, TupleTableSlot *row)
 		segment_info_update(column->segment_info, val, is_null);
 		MemoryContextSwitchTo(row_compressor->per_row_ctx);
```

Comment:
Does this line even serve a purpose anymore, when it immediately switches to `oldcontext` at line 734? Maybe switch to `row_compressor->per_row_ctx->parent` once outside of the for loop, instead of having it switch for every iteration?

Reply:
All memory allocation done inside the for loop should be done as part of per_row_ctx. However as you suggest we can improve by setting right context before the for loop and then switch back to old context once loop is complete.